### PR TITLE
ejb sample update for gradle 2.0 and maven 2.1

### PR DESCRIPTION
**Pull request by cthigh**
@patricktiu I made some changes for the Maven EJB sample as well as adding Gradle. I was finding that by using the parent pom in the root project, Liberty runtimes were being installed in every project and subproject. I also saw your other PR to remove application.xml. Did you intend on merging that PR?

### CODE REVIEW

**Comment by patricktiu** [comment]
I add the parent to the root project and I don't see liberty is installed in ejb-ejb and ejb-war project, when I using Maven command line.  

``` xml
    <parent>
        <groupId>net.wasdev.wlp.maven.parent</groupId>
        <artifactId>liberty-maven-app-parent</artifactId>
        <version>2.1-SNAPSHOT</version>
    </parent>
```

File: README.md [hunk 1]
```diff
@@ -6,34 +6,71 @@ This sample demonstrates injection of an EJB into a servlet. The application con
...
 1. Clone this project and import into Eclipse as an 'Existing Maven Project'.
 2. Right-click the project and select **Run As > Maven Clean**.
 3. Right-click the project and select **Run As > Maven Install**.
 4. Right-click the project and select Run As -> Run on Server.
 5. You should see the following in the console: `Application EJBSample started in XX.XX seconds.`
 
-## Running with Maven
+### Using the command-line with Maven 
 
-This project can be built with Apache Maven. The project uses Liberty Maven Plug-in to automatically download and install Liberty with Java EE7 Web Profile runtime from Maven Central. Liberty Maven Plug-in is also used to create, configure, and run the application on the Liberty server.
```
Comment (patricktiu) [inline README.md:20]: It should be installing `Liberty Java EE 7 Full Platform` runtime

File: README.md [hunk 2]
```diff
@@ -6,34 +6,71 @@ This sample demonstrates injection of an EJB into a servlet. The application con
...
+
+The Liberty Gradle Plug-in supports creating loose application configuration for war tasks, but it does not yet support loose enterprise applications. This example installs the EAR file when building using Gradle. After running the full build, you will see the application installed as `sample.ejb/ejb-ear/build/wlp/usr/servers/ejbServer/apps/ejb-ear.ear`.
+
+### Using the command-line with Gradle 
+
+Use the following steps to build and run the application with Gradle from the root project:
+
+1. Execute the full Gradle build. The Liberty Gradle Plug-in will download and install the Liberty server in the `ejb-ear` project. It will also run all tests.
+    ```bash
+    $ gradle clean build
```
Comment (patricktiu) [inline README.md]: We should use gradle wrapper 
`./gradlew clean build`

**Comment by patricktiu** [comment]
Since you configure the ear plugin to generate the application.xml, you don't need to provide another one in the source. 

``` xml
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-ear-plugin</artifactId>
                <version>2.10.1</version>
                <configuration>
                    <version>7</version>
                    <fileNameMapping>no-version</fileNameMapping>
                    <generateApplicationXml>true</generateApplicationXml>
                </configuration>
</plugin>
```

**Comment by cthigh** [comment]
Talked to @patricktiu on the issue with the parent pom in the root project.  `mvn install` works fine. The problem comes if you issue an `mvn liberty:start-server` and the plugin will create a new default server in the root project.  Of course, this server doesn't have any app installed to it.

**Review by dshimo — APPROVED** [review]
👍

**Review by patricktiu — APPROVED** [review]

**Review by jjvilleg — APPROVED** [review]
Looks great

**Review by mattbsox — APPROVED** [review]

**Comment by d3minem** [comment]
The example is not running. Throws an exception - how to fix it? any hint.

**Browser Output:**

`Error 404: java.io.FileNotFoundException: SRVE0190E: File not found: /ejb-war/ejbservlet/
`

**Exception:**

```
[INFO] [ERROR   ] CWWKE0701E: FrameworkEvent ERROR Bundle:com.ibm.ws.config(id=8) org.osgi.framework.BundleException: Exception in com.ibm.ws.config.internal.WSConfigActivator.start() of bundle com.ibm.ws.config.
[INFO]  at org.eclipse.osgi.internal.framework.BundleContextImpl.startActivator(BundleContextImpl.java:795)
[INFO]  at [internal classes]
[INFO] Caused by: java.lang.NoClassDefFoundError: jdk/internal/reflect/SerializationConstructorAccessorImpl
[INFO]  at java.base/java.lang.ClassLoader.defineClass1(Native Method)
[INFO]  at java.base/java.lang.System$2.defineClass(System.java:2189)
[INFO]  at java.base/jdk.internal.reflect.ClassDefiner.defineClass(ClassDefiner.java:65)
[INFO]  at java.base/jdk.internal.reflect.MethodAccessorGenerator$1.run(MethodAccessorGenerator.java:400)
[INFO]  at java.base/jdk.internal.reflect.MethodAccessorGenerator$1.run(MethodAccessorGenerator.java:394)
[INFO]  at java.base/java.security.AccessController.doPrivileged(AccessController.java:312)
[INFO]  at java.base/jdk.internal.reflect.MethodAccessorGenerator.generate(MethodAccessorGenerator.java:393)
[INFO]  at java.base/jdk.internal.reflect.MethodAccessorGenerator.generateSerializationConstructor(MethodAccessorGenerator.java:112)
[INFO]  at java.base/jdk.internal.reflect.ReflectionFactory.generateConstructor(ReflectionFactory.java:466)
[INFO]  at java.base/jdk.internal.reflect.ReflectionFactory.newConstructorForSerialization(ReflectionFactory.java:458)
[INFO]  at java.base/java.io.ObjectStreamClass.getSerializableConstructor(ObjectStreamClass.java:1582)
[INFO]  at java.base/java.io.ObjectStreamClass$2.run(ObjectStreamClass.java:540)
[INFO]  at java.base/java.io.ObjectStreamClass$2.run(ObjectStreamClass.java:512)
[INFO]  at java.base/java.security.AccessController.doPrivileged(AccessController.java:312)
[INFO]  at java.base/java.io.ObjectStreamClass.<init>(ObjectStreamClass.java:512)
[INFO]  at java.base/java.io.ObjectStreamClass.lookup(ObjectStreamClass.java:403)
[INFO]  at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1143)
[INFO]  at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:353)
[INFO]  at com.ibm.ws.config.admin.internal.ConfigurationStore.serializeConfigurationData(ConfigurationStore.java:213)
[INFO]  ... 1 more
[INFO] Caused by: java.lang.ClassNotFoundException: jdk.internal.reflect.SerializationConstructorAccessorImpl cannot be found by com.ibm.ws.config_1.0.17.cl170220170523-1818
[INFO]  ... 24 more
[INFO]  Event:org.osgi.framework.FrameworkEvent[source=com.ibm.ws.config_1.0.17.cl170220170523-1818 [8]] 
```

**Pull request merged**